[macOS] Yarn executable missing executable bit

Build gives an error:

    /Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(5,5): Error MSB6003: The specified task executable "yarn" could not be run. ApplicationName='/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/dist/bin/yarn', CommandLine=' build', CurrentDirectory='/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS', Native error= Access denied (MSB6003) (StageForge.MacOS)

performing `ls -l packages/Yarn.MSBuild/dist/bin/yarn` shows the permissions `-rw-r-r--r--`; in other words, not executable. Workaround is to run `chmod +x packages/Yarn.MSBuild/dist/bin/yarn`, which fixes the problem temporarily.

I believe a solution is to make sure that the nuget package is built on a unix system, so that the executable bit can survive. If I can be of any help, let me know.

Nevermind, I lied; I forgot to add the yarn reference back after temporarily removing it.

Yarn.MSBuild resolved version is 1.15.2 .

```
$ mono --version
Mono JIT compiler version 5.18.1.28 (2018-08/223ea7ef92e Tue May 21 12:03:39 EDT 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(600)
	Suspend:       preemptive
	GC:            sgen (concurrent by default)
```

Here's a log from running `msbuild`:

```
Microsoft (R) Build Engine version 16.0.459+g5b35a69a2e for Mono
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 6/26/2019 11:01:21 PM.
Project "/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj" on node 1 (default targets).
YarnBuild:
  /Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/dist/bin/yarn run webpack 
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003: The specified task executable "yarn" could not be run. System.ComponentModel.Win32Exception (0x80004005): ApplicationName='/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/dist/bin/yarn', CommandLine=' run webpack', CurrentDirectory='/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS', Native error= Access denied [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at System.Diagnostics.Process.StartWithCreateProcess (System.Diagnostics.ProcessStartInfo startInfo) [0x0029f] in <118907f5eb5f4e538f3e3bdd8361378f>:0  [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at System.Diagnostics.Process.Start () [0x0003a] in <118907f5eb5f4e538f3e3bdd8361378f>:0  [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at (wrapper remoting-invoke-with-check) System.Diagnostics.Process.Start() [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at Microsoft.Build.Utilities.ToolTask.ExecuteTool (System.String pathToTool, System.String responseFileCommands, System.String commandLineCommands) [0x000c9] in <e7918b655c524bee905d411ec2ed37a6>:0  [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at Microsoft.Build.Utilities.ToolTask.Execute () [0x003a8] in <e7918b655c524bee905d411ec2ed37a6>:0  [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
Done Building Project "/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj" (default targets) -- FAILED.

Build FAILED.

"/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj" (default target) (1) ->
(YarnBuild target) -> 
  /Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003: The specified task executable "yarn" could not be run. System.ComponentModel.Win32Exception (0x80004005): ApplicationName='/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/dist/bin/yarn', CommandLine=' run webpack', CurrentDirectory='/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS', Native error= Access denied [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at System.Diagnostics.Process.StartWithCreateProcess (System.Diagnostics.ProcessStartInfo startInfo) [0x0029f] in <118907f5eb5f4e538f3e3bdd8361378f>:0  [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at System.Diagnostics.Process.Start () [0x0003a] in <118907f5eb5f4e538f3e3bdd8361378f>:0  [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at (wrapper remoting-invoke-with-check) System.Diagnostics.Process.Start() [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at Microsoft.Build.Utilities.ToolTask.ExecuteTool (System.String pathToTool, System.String responseFileCommands, System.String commandLineCommands) [0x000c9] in <e7918b655c524bee905d411ec2ed37a6>:0  [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]
/Users/jwostenberg/Code/FSharp/StageForge-WV/packages/Yarn.MSBuild/build/Yarn.MSBuild.targets(10,5): error MSB6003:   at Microsoft.Build.Utilities.ToolTask.Execute () [0x003a8] in <e7918b655c524bee905d411ec2ed37a6>:0  [/Users/jwostenberg/Code/FSharp/StageForge-WV/src/StageForge.MacOS/StageForge.MacOS.fsproj]

    0 Warning(s)
    1 Error(s)
```

I'm referencing it in my Xamarin.Mac project like so:

```
...
  <PropertyGroup>
    <YarnBuildCommand>build</YarnBuildCommand>
  </PropertyGroup>
...
```

--- a/src/Yarn.MSBuild/targets/Yarn.MSBuild.targets
+++ b/src/Yarn.MSBuild/targets/Yarn.MSBuild.targets
@@ -7,6 +7,11 @@
   </PropertyGroup>
 
   <Target Name="YarnBuild" Condition="'$(YarnBuildCommand)' != ''">
+    <!-- Older versions of the NuGet client don't extract yarn with the +x bit -->
+    <Exec Command="chmod +x &quot;$(NodeJsExecutablePath)&quot;; sync"
+          IgnoreExitCode="true"
+          Condition="'$(NodeJsExecutablePath)' != '' AND '$(OS)' != 'Windows_NT'" />
+
     <Yarn Command="$(YarnBuildCommand)"
           WorkingDirectory="$(YarnWorkingDir)"
           ExecutablePath="$(YarnExecutablePath)"

